lib/core/test/unit/Cardano/Wallet/Primitive/ModelSpec.hs

```diff
@@ -819,6 +831,38 @@ prop_applyOurTxToUTxO_someOurs ourState slotNo blockHeight tx utxo =
     shouldHaveResult :: Bool
     shouldHaveResult = evalState (isOurTx tx utxo) ourState
 
+-- Even when transaction processed by 'applyOurTxToUTxO' has some outputs
+-- that are "ours", other outputs that aren't "ours" are of no interest for the
+-- wallet and therefore are ommitted.
+--
+-- This property verifies that all output addresses of applied tx are "ours"
+--
+prop_applyOurTxToUTxO_omitUnrelatedOuputs
+    :: WalletState
+    -> ShowFmt Address
+    -> SlotNo
+    -> Quantity "block" Word32
+    -> Tx
+    -> UTxO
+    -> Property
+prop_applyOurTxToUTxO_omitUnrelatedOuputs st ourAddress slot blocks tx utxo =
+    appliedTx & maybe discard ( \tx' ->
+        report (tx' ^. #outputs) "tx' outputs"  $
+        property $
+            all (`Set.member` ourAddresses st') (address <$> tx' ^. #outputs)
```

Comment:
I think we also want an assertion which says that any output addresses which are in `tx` but not `tx'` must not be members of `ourAddresses`.